pysb/kappa.py

```diff
@@ -289,6 +268,101 @@ def run_kasim(model, time=10000, points=200, output_dir='.', cleanup=False,
+    contact_map : boolean
+        Whether to compute the contact map.
+    output_dir : string
+        The subdirectory in which to generate the Kappa (.ka) file for the
+        model and all output files produced by KaSim. Default value is '.'
+        Note that only relative paths can be specified; paths are relative
+        to the directory where the current Python instance is running.
+        If the specified directory does not exist, an Exception is thrown.
+    cleanup : boolean
+        Specifies whether output files produced by KaSim should be deleted
+        after execution is completed. Default value is False.
+    base_filename : The base filename to be used for generation of the Kappa
+        (.ka) file and all output files produced by KaSim. Defaults to a
+        string of the form::
+
+            '%s_%d_%d_temp' % (model.name, program id, random.randint(0,10000))
+
+        The influence map filename appends '_im.dot' to this base filename; the
+        contact map filename appends '_cm.dot'.
+
+    Returns
+    -------
+    A dict with two entries giving the filenames for the files produced
+
+        * output_dict['im'] gives the influence map filename, or None if not
+          produced
+        * output_dict['cm'] gives the contact map filename, or None if not
+          produced
+    """
+
+    gen = KappaGenerator(model)
+
+    if not base_filename:
+        base_filename = '%s/%s_%d_%d_temp' % (output_dir,
+                        model.name, os.getpid(), random.randint(0, 10000))
+
+    kappa_filename = base_filename + '.ka'
+    im_filename = base_filename + '_im.dot'
+    cm_filename = base_filename + '_cm.dot'
+
+    # Contact map args
+    if contact_map:
+        cm_args = ['--compute-contact-map', '--output-contact-map',
+                   cm_filename]
+    else:
+        cm_args = ['--no-compute-contact-map']
+    # Influence map args
+    if influence_map:
+        im_args = ['--compute-influence-map', '--output-influence-map',
+                   im_filename]
+    else:
+        im_args = ['--no-compute-influence-map']
+    # Full arg list
+    args = [kappa_filename, '--output-directory', output_dir] \
+            + cm_args + im_args
+
+    try:
+        kappa_file = open(kappa_filename, 'w')
+
+        # Generate the Kappa model code from the PySB model and write it to
+        # the Kappa file:
+        kappa_file.write(gen.get_content())
+        kappa_file.close()
+
+        print("Running KaSa")
+        p = subprocess.Popen(['KaSa'] + args)
+                           #stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        p.communicate()
+
+        if p.returncode:
+            raise Exception(p.stdout.read())
+
+    except Exception as e:
+        raise Exception("Problem running KaSa: " + str(e))
+
+    finally:
+        if cleanup:
+            for filename in [kappa_filename, im_filename, cm_filename]:
+                if os.access(filename, os.F_OK):
```

Comment:
Better to just call `unlink` inside a try block, and `pass` on `OSError`. As written this is a race, although we'd probably never see one in practice here.